searchlib/src/vespa/searchlib/common/bitvector.h

```diff
@@ -295,7 +295,7 @@ class BitVector : protected BitWord
     static Index numWords(Index bits) noexcept { return wordNum(bits + 1 + (WordLen - 1)); }
     static Index numBytes(Index bits) noexcept { return numWords(bits) * sizeof(Word); }
     size_t numWords() const noexcept { return numWords(size()); }
-    static constexpr size_t getAlignment() noexcept { return 0x40u; }
+    static constexpr size_t getAlignment() noexcept { return 0x100u; }
```

Comment:
Unfortunately, https://github.com/vespa-engine/vespa/blob/c4e33003e5ce3f385951c107714ede8556ef8083/searchlib/src/vespa/searchlib/diskindex/bitvectordictionary.cpp#L61
indicates that we currently don't store the aligned bitvector byte size  in the file header, thus old bitvector files will be misread when changing the alignment. We likely need a getLegacyFileBytes() method that returns old vector size based on old alignment, used when bit the vector file lacks a file header tag for aligned bitvector byte size.
We should add a file header tag for new bitvector files, and use that value if present instead of the implicit legacy value.

Reply:
Solved as suggested.